nifi-nar-bundles/nifi-ocr-bundle/nifi-ocr-processors/src/main/java/org/apache/nifi/processors/ocr/TesseractOCRProcessor.java

```diff
@@ -0,0 +1,361 @@
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_UNSUPPORTED_IMAGE_FORMAT);
+        relationships.add(REL_ORIGINAL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+
+        List<PropertyDescriptor> descriptorsNew = new ArrayList<>();
+
+        descriptorsNew.add(TESS_DATA_PATH);
+        descriptorsNew.add(new PropertyDescriptor.Builder()
+                .fromPropertyDescriptor(TESSERACT_LANGUAGE)
+                .allowableValues(SUPPORTED_LANGUAGES)
+                .build());
+        descriptorsNew.add(TESSERACT_PAGE_SEG_MODE);
+        descriptorsNew.add(TESSERACT_CONFIGS);
+
+        return descriptorsNew;
+    }
+
+    @Override
+    public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
+        super.onPropertyModified(descriptor, oldValue, newValue);
+
+        if (descriptor.equals(TESS_DATA_PATH)) {
+            getLogger().debug("Tesseract Install path was changed. Building list of supported languages");
+            SUPPORTED_LANGUAGES.clear();
+            SUPPORTED_LANGUAGES.add("eng");
+            //File will always exist since the Validator will take care of that.
+            File[] files = getTesseractLanguages(newValue);
+
+            //Guard against creating an empty list of allowable values in case the user points to an invalid directory
+            if (files != null && files.length > 0) {
+                for (int i = 0; i < files.length; i++) {
+                    getLogger().debug("Found Tesseract supported language: " + files[i].getName());
+                    SUPPORTED_LANGUAGES.add(StringUtils.split(files[i].getName(), ".")[0]);
+                }
+            } else {
+                getLogger().debug("No languages found in user specified Tessdata directory: '" + newValue + "'");
+            }
+
+        }
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        //Setup the Tesseract instance once the processor is scheuled
+        tessInstance = new Tesseract();
+    }
+
+
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        //Transfer the original
+        session.transfer(session.clone(flowFile), REL_ORIGINAL);
+        AtomicBoolean errors = new AtomicBoolean(false);
+
+        FlowFile ff = session.write(flowFile, new StreamCallback() {
+            @Override
+            public void process(InputStream inputStream, OutputStream outputStream) throws IOException {
+                tessInstance.setLanguage(context.getProperty(TESSERACT_LANGUAGE).getValue());
```

Comment:
Since the above is not evaluating expression, could it be set once (e.g., in @OnScheduled)?